integration/Jenkinsfiles/sonar.groovy

```diff
@@ -0,0 +1,67 @@
+/*
+ * (C) Copyright 2017 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     jcarsique
+ */
+
+ node('SLAVE') {
+    tool type: 'ant', name: 'ant-1.9'
+    tool type: 'hudson.model.JDK', name: 'java-8-oracle'
+    tool type: 'hudson.tasks.Maven$MavenInstallation', name: 'maven-3'
+    timeout(time: 5, unit: 'HOURS') {
+        timestamps {
+            stage('clone') {
+                checkout(
+                    [$class: 'GitSCM',
+                        branches: [[name: '*/${BRANCH}']],
```

Comment:
params.BRANCH is safer, we already had issues with parameter typing when directly using BRANCH
ref: https://jenkins.io/doc/book/pipeline/jenkinsfile/#parameters